### PR TITLE
feat: Sticky group headers on overview page

### LINKED ISSUE #15897: Change Overview structure, that subgroups are inside the parent group div to enable easier stickyness

### LINKED ISSUE #15898: Change CSS to enable sticky header

### PR DESCRIPTION
**Pull request by steinsebastian**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html [hunk 1]
```diff
@@ -24,61 +24,128 @@ <h2>{{ 'configurator.overviewForm.noAttributeHeader' | cxTranslate }}</h2>
...
+    <ng-container *ngFor="let group of overviewGroups">
+      <div
+        class="cx-group"
+        [class.topLevel]="level === 1"
+        [class.subgroupTopLevel]="level === 1 && group.subGroups?.length > 0"
+        [class.subgroup]="level > 1"
+        [class.subgroupLevel2]="level === 2"
+        [class.subgroupLevel3]="level === 3"
+        [class.subgroupLevel4]="level === 4"
+        [class.subgroupLevel5]="level === 5"
```
Comment (schuppell) [inline feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html]: Can we make the assignment of the subgrouplevel class as a method?
This would make it easier to change the amount of subgroups that get stacked.

File: feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss [hunk 2]
```diff
@@ -15,127 +15,314 @@
...
+          top: 112px;
+          z-index: -3;
+        }
+      }
 
-        @include media-breakpoint-up(md) {
-          &:not(:first-of-type) {
+      &.subgroupLevel4 {
+        h2 {
+          top: 164px;
```
Comment (schuppell) [inline feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss]: Can we use variables for this?
Does this work, when I change the default font size of my device?

Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss]: I will introduce variables.
Yes, it also works if the size is changed
Screenshot 2022-07-29 at 09 12 58
.

Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss]: Done

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html (hunk 1 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html]: Extracted the style class determination to a method.

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.spec.ts [hunk 3]
```diff
@@ -412,4 +425,56 @@ describe('ConfigurationOverviewFormComponent', () => {
       );
     });
   });
+
+  describe('getGropuLevelStyleClasses', () => {
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.spec.ts]: Typo: should be `getGroupLevelStyleClasses` ;)

File: feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss [hunk 4]
```diff
@@ -15,127 +15,319 @@
     padding-inline-end: 20px;
   }
 
-  .cx-group {
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/styles/_configurator-overview-form.scss:18]: There is a number of styling changes. 
Is there any task to cover them, for instance see `css-changes-in-version-5.md`?

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html [hunk 5]
```diff
@@ -24,61 +24,119 @@ <h2>{{ 'configurator.overviewForm.noAttributeHeader' | cxTranslate }}</h2>
...
+          <span>{{ group.groupDescription }}</span>
+        </h2>
+
+        <div
+          *ngFor="let attributeOverview of group.attributes; let i = index"
+          class="cx-attribute-value-pair"
+          [ngClass]="getStyleClasses(group.attributes, i)"
+        >
+          <ng-container [ngSwitch]="attributeOverview?.type">
+            <ng-container *ngSwitchCase="attributeOverviewType.GENERAL">
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html:109]: This switch-case, `attributeOverviewType.GENERAL`, is equal to default.
It is possible to simplify it, namely to use `*ngSwitchCase="attributeOverviewType.BUNDLE"` case and default?

File: feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html [hunk 6]
```diff
@@ -24,61 +24,119 @@ <h2>{{ 'configurator.overviewForm.noAttributeHeader' | cxTranslate }}</h2>
   <span class="cx-visually-hidden">
     {{ 'configurator.a11y.listOfAttributesAndValues' | cxTranslate }}
   </span>
-  <ng-container *ngFor="let group of overviewGroups">
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.html:27]: There is a number of changes in HTML tree. 
Is there any task to cover them, for instance see `5_0.md`?

**Review by Larisa-Staroverova — CHANGES_REQUESTED** [review]
Please, see my comments

**Review by Larisa-Staroverova — DISMISSED** [review]

**Review by Larisa-Staroverova — DISMISSED** [review]

File: projects/storefrontapp/src/app/app.module.ts [hunk 7]
```diff
@@ -81,6 +82,12 @@ if (!environment.production) {
         level: '4.2',
       },
     }),
+    provideConfig(<ConfiguratorUISettingsConfig>{
```
Comment (ChristophHi) [inline projects/storefrontapp/src/app/app.module.ts]: Please revert this change!

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: projects/storefrontapp/src/app/app.module.ts (hunk 7 above)
Comment (Larisa-Staroverova) [reply projects/storefrontapp/src/app/app.module.ts]: Changes have been reverted

**Review by Larisa-Staroverova — APPROVED** [review]

**Review by ChristophHi — APPROVED** [review]
